### PR TITLE
Adding Metric Specification in Dapr Configuration

### LINKED ISSUE #1643: Add ability to disable metrics through configuration
## In what area(s)?

/area runtime

## Describe the feature

Add metrics disable annotation for perf test

### PR DESCRIPTION
**Pull request by shalabhms**
# Description

Adding Metric Specification to enable/disable the metrics similar to Tracing Configuration.

So by default,  metrics is turned on. Only when user turns off metrics in configuration, Dapr will not generate the metrics.

e.g. 

apiVersion: dapr.io/v1alpha1
kind: Configuration
metadata:
  name: tracing
  namespace: default
spec:
  tracing:
    samplingRate: "1"
  **metric:
    enabled: false**

## Issue reference

Closes #1643 

## Checklist

Please make sure you've  completed the relevant tasks for this PR, out of the following list:

* [x] Code compiles correctly
* [x] Created/updated tests
* [x] Unit tests passing
* [x] End-to-end tests passing
* [x] Extended the documentation / Created issue in the https://github.com/dapr/docs/issues/778
* [ ] Specification has been updated / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Provided sample for the feature / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_

### CODE REVIEW

**Comment by yaron2** [comment]
Great PR!

As we discussed, tracing and metrics should both be consistent with each other and turned on by default.
Having the ability to turn metrics off is what the original issue was about.

**Comment by shalabhms** [comment]
> Great PR!
> 
> As we discussed, tracing and metrics should both be consistent with each other and turned on by default.
> Having the ability to turn metrics off is what the original issue was about.

Thanks @yaron2 . Ok sure, we can keep the default "on" as current behavior.

 I picked up from the issue https://github.com/dapr/dapr/issues/1643 where @youngbupark  mentioned in the discussion comment about perf degradation and default "off" behaviour.

**Comment by youngbupark** [comment]
@yaron2 @shalabhms do we still need  ` -enable-metrics option`?

**Comment by shalabhms** [comment]
> @yaron2 @shalabhms do we still need ` -enable-metrics option`?

No, I am inclined to remove this as well.

**Comment by yaron2** [comment]
Needs to be removed.

**Comment by yaron2** [comment]
Hey @shalabhms, where are we with this?

**Comment by shalabhms** [comment]
> Hey @shalabhms, where are we with this?

@yaron2 , there were some clarification on initing metric exporter. I discussed with @youngbupark . I will be updating the PR shortly.

**Comment by shalabhms** [comment]
Tracking doc issue to update the documentation: dapr/docs#778

**Comment by shalabhms** [comment]
Interesting that added unit test is passing in VS Code way and failing when executed through command line way. Looking into it.

**Comment by shalabhms** [comment]
@yaron2 , @youngbupark , please review as PR is ready to review now.

File: charts/dapr/charts/dapr_operator/templates/dapr_operator_deployment.yaml [hunk 1]
```diff
@@ -83,11 +83,11 @@ spec:
         - "--log-as-json"
 {{- end }}
 {{- if eq .Values.global.prometheus.enabled true }}
-        - "--enable-metrics"
+        - "--enable-metrics-exporter"
```
Comment (yaron2) [inline charts/dapr/charts/dapr_operator/templates/dapr_operator_deployment.yaml:86]: Why did we add -exporter? I think just --enable-metrics is more clear.

Comment (shalabhms) [reply charts/dapr/charts/dapr_operator/templates/dapr_operator_deployment.yaml:86]: @yaron2 , enable-metrics flag is actually starting the exporter and not actually enabling/init the metrics. Till now, metrics is default turned on, metrics is getting enabled under the same flag. So I discussed with @youngbupark  to make a change in the name as well.  I added "-exporter" for clarity but I am open to keep that way also.  @youngbupark , your thoughts ?

Comment (yaron2) [reply charts/dapr/charts/dapr_operator/templates/dapr_operator_deployment.yaml:86]: ok, this is for the operator settings right? I'm fine to keep it as is.

Comment (shalabhms) [reply charts/dapr/charts/dapr_operator/templates/dapr_operator_deployment.yaml:86]: Yes, this one you pointed out is for operator settings.

**Comment by shalabhms** [comment]
@wcs1only , are you on this issue https://github.com/dapr/dapr/issues/2034? . E2E tests failing  consistently in my case.

**Review by youngbupark — APPROVED** [review]

**Comment by shalabhms** [comment]
Tests failed due to external reasons in publishing image time.

**Review by youngbupark — APPROVED** [review]

**Pull request merged**